modules/cert-manager-acme-dns01-aws.adoc

```diff
@@ -0,0 +1,170 @@
+// Module included in the following assemblies:
+//
+// * security/cert_manager_operator/cert-manager-operator-issuer-acme.adoc
+
+:_content-type: PROCEDURE
+[id="cert-manager-acme-dns01-aws_{context}"]
+= Configuring an ACME issuer to solve DNS-01 challenges
+
+You can use {cert-manager-operator} to set up an ACME issuer to solve DNS-01 challenges. This procedure uses _Let's Encrypt_ as the ACME CA server and shows how to solve DNS-01 challenges with Amazon Route 53.
+
+[NOTE]
+====
+It is not supported to use private DNS zones.
+====
+
+.Prerequisites
+
+* You have access to the {product-title} cluster as a user with the `cluster-admin` role.
+* You have set up an IAM role for Amazon Route 53. For more information, see link:https://cert-manager.io/docs/configuration/acme/dns01/route53/[Route53] in the upstream cert-manager documentation.
++
+[NOTE]
+====
+For the {cert-manager-operator}, you must provide an `accessKeyID` and `secretAccessKey` to specify the credentials.
```

Comment:
This is not fully true. CC @anjaltelang @thejasn . I was not sure of this until yesterday on the weekend I had enough time to finally investigate https://cert-manager.io/docs/configuration/acme/dns01/route53/#eks-iam-role-for-service-accounts-irsa and successfully tried in OCP AWS STS env https://docs.openshift.com/container-platform/4.12/authentication/managing_cloud_provider_credentials/cco-mode-sts.html ; in OCP AWS STS env, `accessKeyID` and `secretAccessKey` are not a "must". Let me add my steps and results I tried in my test doc.
```suggestion
For the {cert-manager-operator}, you must provide explicit `accessKeyID` and `secretAccessKey` to specify the credentials unless your cluster is configured to use AWS STS, in which case you can use implicit ambient credentials (e.g. inferred from the environment).
```

Reply:
We can definitely call out that these steps are for non STS clusters.

Reply:
@thejasn Which of the steps in this procedure are for non-STS clusters only?

Reply:
Pretty much all steps across the board were given for non STS clusters only (some may just work on STS but that was not what we had in mind when we gave the steps). We can pull in @xingxingxia's suggestion.

Reply:
Agree

Reply:
Thanks, reworded some. Let me know if it looks okay:

```
If your cluster is _not_ configured to use the AWS Security Token Service (STS), you must must provide explicit `accessKeyID` and `secretAccessKey` credentials. If you cluster uses AWS STS, you can use implicit ambient credentials.
```